File: code/0842/842排列数字.cpp
#include <algorithm>
#include <cstring>
#include <iostream>
using namespace std;

const int N = 10;
int g[N];
bool st[N];
int n;

void dfs(int u) {
  if (u == n) {
    for (int i = 0; i < n; i++)
      cout << g[i] << " ";
    puts("");
    return;
  }

  for (int i = 1; i <= n; i++) {
    if (!st[i]) {
      g[u] = i;
      st[i] = true;
      dfs(u + 1);
      st[i] = false;
    }
  }
}

int main() {
  cin >> n;
  dfs(0);

  return 0;
}

File: code/0842/842排列数字.go
package main

import "fmt"

const N = 10

var g [N]int
var st [N]bool

func dfs(u, n int) {
	if u == n {
		for i := 0; i < n; i++ {
			fmt.Printf("%d ", g[i])
		}
		fmt.Println("")
		return
	}

	for i := 1; i <= n; i++ {
		if st[i] == false {
			st[i] = true
			g[u] = i
			dfs(u+1, n)
			st[i] = false
		}
	}
}

func main() {
	var n int
	fmt.Scan(&n)
	dfs(0, n)
}
